### PR TITLE
Add dedicated scss config files

### PR DESCRIPTION
**Pull request by lucasvallenet**
## Description:
Create dedicated config files.
Use of lists of variables for clarity and standardisation (based on `config/_config.colors.scss`)
Move specific functions to config files.

**Changes :**
- Breakpoints `_config.breakpoints.scss`
- Easing `_config.eases.scss`
- Fonts `_config.fonts.scss`
- Z-indexes `_config.indexes.scss`
- Timing (speed) `_config.timings.scss`

## Breakpoints:
List of breakpoints and media queries function (See [PR #116](https://github.com/locomotivemtl/locomotive-boilerplate/pull/116))
Legacy variables available (`$from-small`, `$to-medium`, ...)

**Usage:**
```
.c-box {
    @media (min-width: $from-small) {}
    // or
    @media #{mq-min("small")} {}

    @media (max-width: $to-small) {}
    // or
    @media #{mq-max("small")} {}

    @media (min-width: $from-small) and (max-width: $to-medium) {}
    // or
    @media #{mq-between("small", "medium)} {}
}
```

## Easing
List of eases and new `ease()` function.

**Usage:**
```
.c-box {
    transition-timing-function: ease(power2-out);
}
```

## Fonts
No breaking changes. 
Moved the `tools/_fonts.scss` to `config/_config.fonts.scss`

## Z-indexes
No breaking changes. 
Moved the `z()` function from `tools/_functions.scss` to `config/_config.indexes.scss`
Moved the `$z-indexes` list from `config/_config.scss` to `config/_config.indexes.scss`

## Timing
List of timing and new `t()` function.

**Usage:**
```
.c-box {
    transition-duration: t(fast);
}
```

### CODE REVIEW

File: assets/styles/settings/_config.indexes.scss
Comment (mcaskill) [inline assets/styles/settings/_config.indexes.scss:1]: I suggest naming this file `_config.zindexes.scss` or maybe `_config.stacking.scss` (referring to the [Stacking Context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context))

File: assets/styles/settings/_config.eases.scss [hunk 1]
```diff
@@ -2,47 +2,74 @@
...
+// Returns ease curve.
+//
+// ```scss
+// .c-box {
+//     transition-timing-function: ease(power2-out);
+// }
+// ```
+//
+// @param  {string} $key - The ease key in $eases.
+// @return {bezier curve}
```
Comment (mcaskill) [inline assets/styles/settings/_config.eases.scss:67]:
```suggestion
// @return {easing-function}
```

File: assets/styles/settings/_config.fonts.scss [hunk 2]
```diff
@@ -59,3 +93,12 @@
...
 }
+
+
+// Include fonts
+// ==========================================================================
+
+// Font directory
+$font-dir: "../fonts/";
+
+@include font-faces($font-faces, $font-dir);
```
Comment (mcaskill) [inline assets/styles/settings/_config.fonts.scss:104]: I think this should remain in `_document.scss`. The configuration files, like tools, should not output CSS.

**Review by mcaskill — CHANGES_REQUESTED** [review]
As long as function declarations in configuration files is limited to accessor functions (to retrieve a value from a variable), I'm fine with this proposed change.

File: assets/styles/settings/_config.indexes.scss
Comment (lucasvallenet) [reply assets/styles/settings/_config.indexes.scss:1]: Done !

File: assets/styles/settings/_config.eases.scss (hunk 1 above)
Comment (lucasvallenet) [reply assets/styles/settings/_config.eases.scss]: Done !

File: assets/styles/settings/_config.fonts.scss (hunk 2 above)
Comment (lucasvallenet) [reply assets/styles/settings/_config.fonts.scss]: True, done !

**Review by mcaskill — APPROVED** [review]

**Review by mcaskill — APPROVED** [review]

**Comment by lucasvallenet** [comment]
@mcaskill I was not able to purge the css for the spacing properties

**Comment by mcaskill** [comment]
> I was not able to purge the css for the spacing properties

@devenini I believe you have more experience with PurgeCSS.

**Review by mcaskill — APPROVED** [review]

**Pull request merged**